google/cloud/storage/internal/grpc_bucket_request_parser.cc

```diff
@@ -170,6 +170,50 @@ NativeIamPolicy GrpcBucketRequestParser::FromProto(
   return result;
 }
 
+google::iam::v1::SetIamPolicyRequest GrpcBucketRequestParser::ToProto(
+    SetNativeBucketIamPolicyRequest const& request) {
+  google::iam::v1::SetIamPolicyRequest result;
+  result.set_resource("projects/_/buckets/" + request.bucket_name());
+  auto& policy = *result.mutable_policy();
+  policy.set_version(request.policy().version());
+  policy.set_etag(request.policy().etag());
+  for (auto const& b : request.policy().bindings()) {
+    google::iam::v1::Binding binding;
+    binding.set_role(b.role());
+    for (auto const& m : b.members()) binding.add_members(m);
+    if (b.has_condition()) {
+      auto& expr = *binding.mutable_condition();
+      expr.set_expression(b.condition().expression());
+      expr.set_location(b.condition().location());
+      expr.set_title(b.condition().title());
+      expr.set_description(b.condition().description());
+    }
+    *policy.add_bindings() = std::move(binding);
+  }
+  return result;
+}
+
+// TODO(#5929) - remove after decommission is completed
+#include "google/cloud/internal/disable_deprecation_warnings.inc"
+
+google::iam::v1::SetIamPolicyRequest GrpcBucketRequestParser::ToProto(
+    SetBucketIamPolicyRequest const& request) {
+  google::iam::v1::SetIamPolicyRequest result;
+  result.set_resource("projects/_/buckets/" + request.bucket_name());
+  auto& policy = *result.mutable_policy();
+  policy.set_version(request.policy().version);
+  policy.set_etag(request.policy().etag);
+  for (auto const& b : request.policy().bindings) {
+    google::iam::v1::Binding binding;
+    binding.set_role(b.first);
+    for (auto const& m : b.second) binding.add_members(m);
+    *policy.add_bindings() = std::move(binding);
+  }
+  return result;
+}
+
+#include "google/cloud/internal/diagnostics_pop.inc"
```

Comment:
can we add the same `TODO(#5929)` here, to make super sure we won't forget to remove the line.

Reply:
We really need to remove the whole function, I changed the TODO comment to say that.